internal/connect/command_offset_describe.go

```diff
@@ -0,0 +1,173 @@
+package connect
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/tidwall/pretty"
+
+	connectv1 "github.com/confluentinc/ccloud-sdk-go-v2/connect/v1"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/examples"
+	"github.com/confluentinc/cli/v3/pkg/kafka"
+	"github.com/confluentinc/cli/v3/pkg/output"
+	"github.com/confluentinc/cli/v3/pkg/retry"
+)
+
+type humanOffsetConnectorOut struct {
+	Id       string `human:"ID"`
+	Name     string `human:"Name"`
+	Offsets  string `human:"Offsets"`
+	Metadata string `human:"Metadata"`
+}
+
+type serializedOffsetConnectorOut struct {
+	Id       string                                      `json:"id" yaml:"id"`
+	Name     string                                      `json:"name" yaml:"name"`
+	Offsets  []map[string]any                            `json:"offsets" yaml:"offsets"`
+	Metadata connectv1.ConnectV1ConnectorOffsetsMetadata `json:"metadata" yaml:"metadata"`
+}
+
+func (c *offsetCommand) newDescribeCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "describe <id>",
+		Short:             "Describe connector offsets.",
+		Args:              cobra.ExactArgs(1),
+		RunE:              c.describe,
+		ValidArgsFunction: pcmd.NewValidArgsFunction(c.validArgs),
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Describe offsets for a connector in the current or specified Kafka cluster context.",
+				Code: "confluent connect offset describe lcc-123456",
+			},
+			examples.Example{
+				Code: "confluent connect offset describe lcc-123456 --cluster lkc-123456",
+			},
+		),
+	}
+
+	cmd.Flags().Int32("staleness-threshold", 120, "Repeatedly fetches offsets, until we get an offset with an observed time within staleness threshold in seconds. (min 5)")
+	cmd.Flags().Int32("refetch-timeout", 30, "Max time in seconds to wait until we get an offset within the staleness threshold.")
```

Comment:
Correct me if I'm wrong, but if `--refetch-timeout` < `--staleness-threshold` (this is the default), there are two cases:
1. The last offsets are under the staleness threshold and the command returns.
2. The last offsets are above the staleness threshold, and we wait at most 30s to get offsets with a staleness threshold under 120s.

In case (2), won't the offsets have a staleness threshold of at most 30s since that's how long we're waiting? I don't see how in either case the value of `--staleness-threshold` would matter if it's greater than `--refetch-timeout`

Reply:
it 2nd case, the 2nd attempt will give us the right value. and mostly we would expect staleness-threshold (if set by customer) to be less that refetch timeout.